src/v/raft/consensus_client_protocol.h

```diff
@@ -27,22 +27,20 @@ class consensus_client_protocol final {
 public:
     struct impl {
         virtual ss::future<result<vote_reply>>
-        vote(model::node_id, vote_request&&, rpc::client_opts) = 0;
+        vote(vnode, vote_request&&, rpc::client_opts) = 0;
```

Comment:
We should include target vnode into every request (`vote_request`, `append_entries_request`, `heartbeat_request`, `install_snapshot_request`, `timeout_now_request`) and check on arrival that the recipient's vnode matches the target vnode. Otherwise we risk to do a cross revision rpc and violate raft's assumptions. The same applies to the replies too.